### PR TITLE
Don't preemptively set u+x permissions when executing a process

### PR DESCRIPTION
**Pull request by almiliMSFT**
[AB#1488150](https://dev.azure.com/mseng/708e929f-6bd5-415a-8daf-25b1dac08dd8/_workitems/edit/1488150)

[AB#1516419](https://dev.azure.com/mseng/708e929f-6bd5-415a-8daf-25b1dac08dd8/_workitems/edit/1516419)

### CODE REVIEW

File: Public/Src/Engine/Processes/UnSandboxedProcess.cs [hunk 1]
```diff
@@ -301,16 +301,19 @@ protected Process CreateAndSetUpProcess()
         {
             Contract.Requires(Process == null);
 
-            if (!FileUtilities.FileExistsNoFollow(ProcessInfo.FileName))
+            var mode = GetFilePermissionsForFilePath(ProcessInfo.FileName, followSymlink: false);
```
Comment (mpysson) [inline Public/Src/Engine/Processes/UnSandboxedProcess.cs]: >GetFilePermissionsForFilePath [](start = 23, length = 29)

Does this need to be within the #if PLATFORM_OSX?

File: Public/Src/Engine/Processes/UnSandboxedProcess.cs [hunk 2]
```diff
@@ -301,16 +301,19 @@ protected Process CreateAndSetUpProcess()
         {
             Contract.Requires(Process == null);
 
-            if (!FileUtilities.FileExistsNoFollow(ProcessInfo.FileName))
```
Comment (mpysson) [inline Public/Src/Engine/Processes/UnSandboxedProcess.cs]: >if (!FileUtilities.FileExistsNoFollow(ProcessInfo.FileName)) [](start = 12, length = 60)

I don't think the new behavior is equivalent in the windows case

File: Public/Src/Engine/Processes/UnSandboxedProcess.cs (hunk 1 above)
Comment (almiliMSFT) [reply Public/Src/Engine/Processes/UnSandboxedProcess.cs]: You are right about both, will fix it...

File: Public/Src/Engine/Processes/UnSandboxedProcess.cs [hunk 3]
```diff
@@ -301,16 +301,24 @@ protected Process CreateAndSetUpProcess()
         {
             Contract.Requires(Process == null);
 
+#if !PLATFORM_OSX
```
Comment (dannyvv) [inline Public/Src/Engine/Processes/UnSandboxedProcess.cs]: W
Shouldn't we keep the exe does not exist check for windows?

Comment (almiliMSFT) [reply Public/Src/Engine/Processes/UnSandboxedProcess.cs]: It is kept (here it's `! PLATFORM_OSX` so this executes on Windows)

**Pull request merged**